src/components/PoolModal/LiquidityCard.tsx

```diff
@@ -0,0 +1,170 @@
+import { getTokenAddress, Token, tokenAddressToToken } from "@/lib/Token";
+import { useEffect, useState } from "react";
+import { twMerge } from "tailwind-merge";
+import { SolidButton } from "@/components/SolidButton";
+import { TokenSelector } from "@/components/TokenSelector";
+import { LAMPORTS_PER_SOL, PublicKey } from "@solana/web3.js";
+import { useConnection, useWallet } from "@solana/wallet-adapter-react";
+import { getAssociatedTokenAddress } from "@solana/spl-token";
+import { Pool, usePools } from "@/hooks/usePools";
+import { SidebarTab } from "../SidebarTab";
+
+import Add from "@carbon/icons-react/lib/Add";
+import Subtract from "@carbon/icons-react/lib/Subtract";
+import { LpSelector } from "./LpSelector";
+import { changeLiquidity } from "src/actions/changeLiquidity";
+
+interface Props {
+  className?: string;
+  pool: Pool;
+  //   amount: number;
+  //   token: Token;
+  //   onChangeAmount?(amount: number): void;
+  //   onSelectToken?(token: Token): void;
+}
+
+enum Tab {
+  Add,
+  Remove,
+}
+
+export default function LiquidityCard(props: Props) {
+  const [payToken, setPayToken] = useState(Token.SOL);
+  const [tokenAmount, setTokenAmount] = useState(0.2);
+
+  const [tab, setTab] = useState(Tab.Remove);
+
+  const [payTokenBalance, setPayTokenBalance] = useState(0);
+  const [liqBalance, setLiqBalance] = useState(0);
+  const [liqAmount, setLiqAmount] = useState(1);
+
+  const { wallet, publicKey, signTransaction } = useWallet();
+  const { connection } = useConnection();
+
+  let tokenList = Object.keys(props.pool?.tokens).map((token) => {
+    return tokenAddressToToken(token);
+  });
+
+  useEffect(() => {
+    async function fetchData() {
+      if (payToken === Token.SOL) {
+        let balance = await connection.getBalance(publicKey);
+        setPayTokenBalance(balance / LAMPORTS_PER_SOL);
```

Comment:
lets use wrapper toTokenAmount / toUiAmount for all manipulations like this. Number of decimal places to compute the multiplier or divisor should be taken from the tokens DB and not hard-coded